CMakeLists.txt

```diff
@@ -32,6 +32,14 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/")
 
 include(CTest)
 
+option(USE_SYSTEM_DATE "\
+    Use find_package to find installed HowardHinnant's \
+    date library instead of fetching it from github" OFF
+)
+
+if(USE_SYSTEM_DATE)
+    find_package(date REQUIRED)
+endif()
 
 ### Dependencies 
 add_subdirectory(dependencies)
```

Comment:
You should put the add_subdirectory here in the else I suppose :-) Otherwise you have not much gained, it will still fetch date. 
And could you move the comment in line 44 above that "### Dependencies". That was there to introduce the section for dependencies, so the cmake file will be clean

Reply:
> You should put the add_subdirectory here in the else 

Yes, but there is a little inconsistency here: in your case, if not `USE_SYSTEM_DATE`, we `add_subdirectory(dependencies)`, which potentially might contain not only date dependency.

> Otherwise you have not much gained, it will still fetch date.

It will not - I wrapped dependencies/CMakeLists.txt content into `if(NOT USE_SYSTEM_DATE)` block. I actually don't like this variant, but putting `add_subdirectory(dependencies)` in else branch looks a bit illogical too.

Maybe you could suggest an approach that will avoid the mentioned problems?

Reply:
Ah right, sorry, I didn't see the other change. I think this is fine. Usually in my projects I use two ifs, one around the fetchContent_Declare and one around the add_subdirectory(<NameOfDepedency>). The reason is that with multiple dependencies you first specify all FetchContent_Declare and then all add_subdirectory / FetchContent_MakeAvailable, so common dependencies are only used one. 

But as we only have one dependency right now, it doesn't matter I suppose. If there will be a second one we can split this up further.